pdns/dnsdistdist/dnsdist-kvs.cc

```diff
@@ -30,15 +30,24 @@ std::vector<std::string> KeyValueLookupKeySourceIP::getKeys(const ComboAddress&
   std::vector<std::string> result;
   ComboAddress truncated(addr);
 
+  std::string key;
   if (truncated.isIPv4()) {
     truncated.truncate(d_v4Mask);
-    result.emplace_back(reinterpret_cast<const char*>(&truncated.sin4.sin_addr.s_addr), sizeof(truncated.sin4.sin_addr.s_addr));
+    key.reserve(sizeof(truncated.sin4.sin_addr.s_addr) + (d_includePort ? sizeof(truncated.sin4.sin_port) : 0));
+    key.append(reinterpret_cast<const char*>(&truncated.sin4.sin_addr.s_addr), sizeof(truncated.sin4.sin_addr.s_addr));
   }
   else if (truncated.isIPv6()) {
     truncated.truncate(d_v6Mask);
-    result.emplace_back(reinterpret_cast<const char*>(&truncated.sin6.sin6_addr.s6_addr), sizeof(truncated.sin6.sin6_addr.s6_addr));
+    key.reserve(sizeof(truncated.sin6.sin6_addr.s6_addr) + (d_includePort ? sizeof(truncated.sin4.sin_port) : 0));
```

Comment:
The ports are only useful for IPv4 CG-NAT setups, not for IPv6. And I guess `sin4.sin_port` would need to use `sin6`?

Reply:
Right, I ponder adding it only for IPv4 but it seemed easier to have the same code path, even though it will probably never be used for IPv6. The structures are designed in such a way that accessing `sin4.sin_port` or `sin6.sin6_port` is equivalent.